### PR TITLE
Adding stacking to the charts

### PR DESCRIPTION
**Pull request by bulyonov**
Stacking was added to the orthogonal chart. It can be enabled using `stackBy` parameter on `XYPlot`: 

```jsx
<XYPlot stackBy="y">
```

Some of the series (area series and column series) have to be updated to support the new behavior.

### CODE REVIEW

File: src/lib/plot/xy-plot.js [hunk 1]
```diff
@@ -225,11 +216,45 @@ class XYPlot extends React.Component {
...
-    const {width, height, animation} = this.props;
+  /**
+   * Checks if the plot is empty or not.
+   * Currently checks the data only.
+   * @returns {boolean} True for empty.
+   * @private
+   */
+  _isPlotEmpty() {
+    const {data} = this.state;
+    return !data || !data.length || ![].concat(...data).filter(d => d).length
```
Comment (chrisirhc) [inline src/lib/plot/xy-plot.js]: I don't think the `[].concat(...data)` is needed, unless `data` has a `length` attribute but doesn't have `.filter` which should break everything anyway.

File: src/lib/utils/series-utils.js [hunk 2]
```diff
@@ -65,6 +65,42 @@ function collectSeriesTypesInfo(children) {
...
+    }
+    childData.push(data.map((d, dIndex) => {
+      let prevValue = 0;
+      let prevIndex = childIndex - 1;
+      while (prevIndex >= 0) {
+        if (childData[prevIndex]) {
+          break;
+        }
+        prevIndex--;
+      }
```
Comment (chrisirhc) [inline src/lib/utils/series-utils.js]: This seems quite confusing, you're looping through the children, then looping to find the previous child that is truthy?

This logic to find the `prevIndex` seems like it's something that could be outside of the `data.map` loop. You're trying to find the previous child's data that is truthy right?

File: src/lib/utils/series-utils.js [hunk 3]
```diff
@@ -65,6 +65,42 @@ function collectSeriesTypesInfo(children) {
   return result;
 }
 
+export function getStackedData(children, attr) {
+  const childData = [];
+  const attr0 = `${attr}0`;
+  children.forEach((child, childIndex) => {
```
Comment (chrisirhc) [inline src/lib/utils/series-utils.js]: Shouldn't this method do nothing if `attr` is falsey?

File: src/test/series-utils.js [hunk 4]
```diff
@@ -22,7 +22,6 @@ import test from 'tape';
 import 'babel-polyfill';
 
 import React from 'react';
-import equal from 'deep-equal';
 
 import {
```
Comment (chrisirhc) [inline src/test/series-utils.js]: I think it's a good time to add a test for the `getStackedData` since it's quite complex looking.

File: src/lib/utils/series-utils.js (hunk 3 above)
Comment (bulyonov) [inline src/lib/utils/series-utils.js]: @chrisirhc, it cannot be falsey.

File: src/lib/plot/xy-plot.js (hunk 1 above)
Comment (bulyonov) [inline src/lib/plot/xy-plot.js]: @chrisirhc , `[].concat(...data)` is needed: it's an array of arrays (e.g. array of data for each series) and I have to know if *any* data is passed to it.

File: src/test/series-utils.js (hunk 4 above)
Comment (bulyonov) [inline src/test/series-utils.js]: @chrisirhc , should be in a separate pull request. Much more changes are pending and blocked by this diff.

File: src/lib/plot/xy-plot.js (hunk 1 above)
Comment (chrisirhc) [inline src/lib/plot/xy-plot.js]: How about `data.some(d => d.some(d => d))` ?

File: src/lib/utils/series-utils.js (hunk 3 above)
Comment (chrisirhc) [inline src/lib/utils/series-utils.js]: line 85 of xy-plot calls this with `stackBy` which could be empty/null .

Comment (bulyonov) [inline src/lib/utils/series-utils.js]: Yes. Even it is empty the correct result is returned (please see `!attr` check on line 78). However, I moved it in the inner function to avoid misunderstanding.

File: src/lib/utils/series-utils.js (hunk 2 above)
Comment (bulyonov) [inline src/lib/utils/series-utils.js]: Done, thank you!
